config/configtelemetry/doc.go

```diff
@@ -4,4 +4,51 @@
 // Package configtelemetry defines various telemetry level for configuration.
 // It enables every component to have access to telemetry level
 // to enable metrics only when necessary.
+//
+// This document provides guidance on which telemetry level to adopt for Collector metrics.
+// When adopting a telemetry level, the developer is expected to rely on this guidance to
+// justify their choice of telemetry level.
+//
+// 1. configtelemetry.None
+//
+// No telemetry data should be collected.
+//
+// 2. configtelemetry.Basic
+//
+// Signals associated with this level cover the essential coverage of the collector telemetry.
+// It should only be used for internal collector telemetry.
+//
+// This is the default level recommended when running the Collector.
+//
+// Signals using this telemetry level can use this guidance:
+// * The signals associated with this level must show low cardinality, the number of combinations of dimension values.
+// * The signals associated with this level must represent a small data volume. Examples follow:
+//   - A max cardinality (total possible combinations of dimension values) of 50.
+//   - At most a span actively recording simultaneously, covering the critical path.
```

Comment:
I don't think we should add this section, given that we don't expect components to emit at this level.

Reply:
sounds good, removing